service/src/internal/inclusion.rs

```diff
@@ -131,38 +132,29 @@ impl InclusionService {
                         .await
                     {
                         Ok(zk_job_id) => {
+                            debug!("Proof request {zk_job_id:?} started");
                             job_status = JobStatus::ZkProofPending(zk_job_id);
                             self.send_job_with_new_status(job_key, job_status, job)?;
                         }
                         Err(e) => {
                             error!("{job:?} failed progressing DataAvailable: {e}");
-                            job_status = JobStatus::Failed(
-                                e,
-                                Some(JobStatus::DataAvailable(proof_input).into()),
-                            );
-                            self.finalize_job(&job_key, job_status)?;
+                            // NOTE: we internally finalize the job in `handle_da_client_error`
                         }
                     };
-                    debug!("ZK request sent");
                 }
                 JobStatus::ZkProofPending(zk_request_id) => {
                     debug!("ZK request waiting");
-                    match self.wait_for_zk_proof(&job_key, zk_request_id).await {
+                    match self.wait_for_zk_proof(&job, &job_key, zk_request_id).await {
                         Ok(zk_proof) => {
                             info!("🎉 {job:?} Finished!");
                             job_status = JobStatus::ZkProofFinished(zk_proof);
                             self.finalize_job(&job_key, job_status)?;
                         }
                         Err(e) => {
                             error!("{job:?} failed progressing ZkProofPending: {e}");
-                            job_status = JobStatus::Failed(
-                                e,
-                                Some(JobStatus::ZkProofPending(zk_request_id).into()),
-                            );
-                            self.finalize_job(&job_key, job_status)?;
+                            // NOTE: we internally finalize the job in `handle_zk_client_error`
```

Comment:
This kinda sucks. I would much rather handle what to do here, but we need to catch on the error in handler :face_exhaling: 
it works, but isn't great.